src/managers/ChannelManager.js

```diff
@@ -4,13 +4,22 @@ const CachedManager = require('./CachedManager');
 const Channel = require('../structures/Channel');
 const { Events, ThreadChannelTypes } = require('../util/Constants');
 
+let cacheWarningEmitted = false;
+
 /**
  * A manager of channels belonging to a client
  * @extends {CachedManager}
  */
 class ChannelManager extends CachedManager {
   constructor(client, iterable) {
     super(client, Channel, iterable);
+    if (!cacheWarningEmitted && this._cache.constructor.name !== 'Collection') {
```

Comment:
Maybe check if `this._cache.constructor` is `Collection`, instead of checking for the name?